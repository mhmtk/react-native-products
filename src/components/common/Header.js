//import libs
import React from 'react';
import { Text, View } from 'react-native';

//make the comp
const Header = (props) => { 
    const { textStyle, viewStyle } = styles;

    return (
    <View style={viewStyle}>
            <Text style={textStyle}>{props.headerText}</Text>
        </View>
    );
};

const styles = {
    textStyle: {
        fontSize: 25
    },
    viewStyle: {
        backgroundColor: '#F8F8F8',
        justifyContent: 'center',
        alignItems: 'center',
        height: 70,
        paddingTop: 20,    
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.2,
        elevation: 2,
        position: 'relative'
    }
};
//make component available to other parts of app
export { Header };
