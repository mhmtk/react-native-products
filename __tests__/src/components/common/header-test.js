import 'react-native';
import React from 'react';
import { Header, Button } from '../../../../src/components/common';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('header renders correctly', () => {
  const tree = renderer.create(<Header headerText="Products" />).toJSON();
  expect(tree).toMatchSnapshot();
});
