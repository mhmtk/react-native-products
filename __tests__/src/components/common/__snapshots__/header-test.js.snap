// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F8F8F8",
      "elevation": 2,
      "height": 70,
      "justifyContent": "center",
      "paddingTop": 20,
      "position": "relative",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 25,
      }
    }
  >
    Products
  </Text>
</View>
`;
